src/snowflake/connector/auth_by_plugin.py

```diff
@@ -37,3 +93,36 @@ def handle_failure(self, ret):
                 "sqlstate": SQLSTATE_CONNECTION_WAS_NOT_ESTABLISHED,
             },
         )
+
+    def handle_timeout(
+        self,
+        authenticator: str,
+        service_name: any,
```

Comment:
You haven't fixed this, please see: https://github.com/snowflakedb/snowflake-connector-python/pull/919#discussion_r741351525

Reply:
Oh yeah about this - the `any` type just signifies that we don't care for the type. It can be _any_ type. I think I should have put everything as _any_ based on that logic. Here's the reference for you https://docs.python.org/3/library/typing.html#typing.Any

Reply:
the fact that we don't care about the type is re-enforced by the first line in the function definition - I delete all the arguments to prevent any accidental usage.

Reply:
Updating to Optional